packages/react-utils/src/referentiallyStable/useReferentiallyStableValue.ts

```diff
@@ -0,0 +1,14 @@
+import deepEqual from 'fast-deep-equal/es6/index.js'
+import { useEffect, useRef } from 'react'
+
+export function useReferentiallyStableValue<T>(value: T): T {
+	const ref = useRef(value)
+
+	useEffect(() => {
+		if (!deepEqual(ref.current, value)) {
+			ref.current = value
+		}
+	}, [value])
+
+	return ref.current = value
```

Comment:
i don't get this. this looks like it ALWAYS returns the input `value`, and also executes the useEffects, which is totally pointless and the ref is not used at all in the end..

Reply:
Removed.